app/models/code_ocean/file.rb

```diff
@@ -60,6 +60,8 @@ class File < ApplicationRecord
     validates :weight, absence: true, unless: :teacher_defined_assessment?
     validates :file, presence: true if :context.is_a?(Submission)
     validates :context_type, inclusion: {in: ALLOWED_CONTEXT_TYPES}
+    validates :path, uniqueness: {scope: %I[name file_type context_id context_type role]}
```

Comment:
Don't we have records that have this issue right now? I am worried that this migration will suddenly stop teachers from editing the file.

Reply:
I'm not sure about any existing records, since I don't have access, but it would indeed be important to check before merging this PR.